### PR TITLE
Fix for a bug when searching for 'develop'

### PR DESCRIPTION
**Pull request by dawelter2**
Fix for a bug found when trying to search for a branch named 'develop', 
Apparently one of the results was missing a key in the dictionary, so instead of access the key, i'm using `dict.get(<key>)` to return None instead of raise KeyError.

### CODE REVIEW

File: test_esss_jenkins.py [hunk 1]
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
...
 
 
+def test_fetch_job_status(jenkins_plugin, mocker):
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild", "result": None})
+    jenkins_plugin._fetch_job_status("dummy")
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild"})
```
Comment (prusse-martin) [inline test_esss_jenkins.py]: Was this the returned value triggering the error?

Comment (nicoddemus) [reply test_esss_jenkins.py]: Yep

File: test_esss_jenkins.py [hunk 2]
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
     assert settings['last_job_listing'] == ['job-A', 'job-B']
 
 
+def test_fetch_job_status(jenkins_plugin, mocker):
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild", "result": None})
+    jenkins_plugin._fetch_job_status("dummy")
```
Comment (nicoddemus) [inline test_esss_jenkins.py]: Assert the return status here please

File: test_esss_jenkins.py [hunk 3]
```diff
@@ -229,6 +229,17 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
...
+    mocker.patch.object(jenkins_plugin,
+                        '_get_jenkins_json_request',
+                        return_value={"_class": "hudson.model.FreeStyleBuild"})
+    jenkins_plugin._fetch_job_status("dummy")
```
Comment (nicoddemus) [inline test_esss_jenkins.py]: Assert the return status here please

**Review by nicoddemus — CHANGES_REQUESTED** [review]

File: dev-requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 -r requirements.txt
-errbot
+errbot~=5.2
 pytest
 pytest-mock
+pluggy==0.11
```
Comment (nicoddemus) [inline dev-requirements.txt]: Put comments here explaining why those pinnings (for pluggy also link to the relevant issue in pluggy's repo)

File: test_esss_jenkins.py [hunk 5]
```diff
@@ -230,14 +231,43 @@ def test_bhist(jenkins_plugin, testbot, mocker, LineMatcher):
 
 
 def test_fetch_job_status(jenkins_plugin, mocker):
```
Comment (nicoddemus) [inline test_esss_jenkins.py]: Please make this test parametrized. 😉

**Review by nicoddemus — CHANGES_REQUESTED** [review]

File: test_esss_jenkins.py (hunk 2 above)
Comment (dawelter2) [reply test_esss_jenkins.py]: Done!

File: test_esss_jenkins.py (hunk 5 above)
Comment (dawelter2) [reply test_esss_jenkins.py]: Done, just couldn't figure out how to parametrize the status "NOT_STARTED", so created a new test.

**Review by prusse-martin — APPROVED** [review]

**Review by nicoddemus — APPROVED** [review]

**Pull request merged**